File: Java Programs/Practice/ArmstrongNumberGenerate.java
import java.util.*;

public class ArmstrongNumberGenerate
{
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);
		
		System.out.print("\n Enter the number of terms : ");
		int n = sc.nextInt();
		
		System.out.print("\n The Armstrong numbers are : ");
		
		int count = 0;
		long num = 0;
		
		while(count != n)
		{
			long temp = num;
			int totalDigits = (int)Math.log10(num) + 1;
			long result = 0;
			
			while(temp != 0)
			{
				int rem = (int)temp % 10;
				result = result + (int)Math.pow(rem, totalDigits);
				temp = temp / 10;
			}
			
			if(result == num)
			{
				System.out.print(num + " ");
				count++;
			}
			
			num++;
		}
		
		System.out.println();
		
		sc.close();
	}
}

File: Python Programs/armstrong_number_generate.py
import math

n = int(input("Enter the no. of terms : "))

print("The armstrong numbers are :", end = " ")

num = 0
count = 0

while(count != n):
    temp = num

    if num == 0:
        total_digits = 1
    else:
        total_digits = int(math.log10(num)) + 1

    result = 0
    
    while(temp != 0):
        rem = temp % 10
        result = result + int(pow(rem, total_digits))
        temp = temp // 10

    if(result == num):
        print(num, end = " ")
        count += 1

    num += 1
